sxw_resource.c

```diff
@@ -300,101 +263,101 @@ static void _transp_contribution_by_group(RealF use_by_group[]) {
+        transp_window->average += (sumTranspTotal - transp_window->transp[transp_window->oldest_index])/transp_window->size;
         //add the new value, subtract the old value from the ratio average;
-        transp_window.ratio_average += (transp_ratio - transp_window.ratios[transp_window.oldest_index])/transp_window.size;
+        transp_window->ratio_average += (transp_ratio - transp_window->ratios[transp_window->oldest_index])/transp_window->size;
         //put the new transpiration in the window
-        transp_window.transp[transp_window.oldest_index] = sumTranspTotal;
+        transp_window->transp[transp_window->oldest_index] = sumTranspTotal;
         //put the new ratio in the window
-        transp_window.ratios[transp_window.oldest_index] = transp_ratio;
+        transp_window->ratios[transp_window->oldest_index] = transp_ratio;
         // calculate the new sum of squares value
-        RealF ssqr = (transp_ratio - transp_window.ratio_average) * (transp_ratio - transp_window.ratio_average);
+        RealF ssqr = (transp_ratio - transp_window->ratio_average) * (transp_ratio - transp_window->ratio_average);
         // add the new sum of squares, subtract the old.
-        transp_window.sum_of_sqrs += ssqr - transp_window.SoS_array[transp_window.oldest_index];
+        transp_window->sum_of_sqrs += ssqr - transp_window->SoS_array[transp_window->oldest_index];
         // update the sum of squares window.
-        transp_window.SoS_array[transp_window.oldest_index] =  ssqr;
+        transp_window->SoS_array[transp_window->oldest_index] =  ssqr;
         //calculate the standard deviation
-        transp_ratio_sd = final_running_sd(transp_window.size, transp_window.sum_of_sqrs);
+        transp_ratio_sd = final_running_sd(transp_window->size, transp_window->sum_of_sqrs);
     }
 
-    //printf("Year %d: ratio = %f, mean = %f, sos = %f sd = %f\n",Globals.currYear,
-              // transp_ratio,transp_window.ratio_average, transp_window.sum_of_sqrs, transp_ratio_sd);
+    //printf("Year %d: ratio = %f, mean = %f, sos = %f sd = %f\n",Globals->currYear,
+              // transp_ratio,transp_window->ratio_average, transp_window->sum_of_sqrs, transp_ratio_sd);
 
     // If this year's transpiration is notably low (1 sd below the mean), add additional transpired water
-    if (transp_ratio < (transp_window.ratio_average - 1 * transp_ratio_sd)) {
-        //printf("Year %d below 1 sd: ratio = %f, average = %f, sd = %f\n", Globals.currYear,transp_ratio,
-                                                          //transp_window.ratio_average, transp_ratio_sd);
-            RealF min = transp_window.ratio_average - transp_ratio_sd;
-            RealF max = transp_window.ratio_average + transp_ratio_sd;
+    if (transp_ratio < (transp_window->ratio_average - 1 * transp_ratio_sd)) {
+        //printf("Year %d below 1 sd: ratio = %f, average = %f, sd = %f\n", Globals->currYear,transp_ratio,
```

Comment:
This line can probably safely be removed.

Reply:
Resolved by 103ca00.